### ISSUE TITLE
Liveness of this project

### ISSUE THREAD

**Issue by javadba** [mpastell/Pweave#140]
Looks like a great project.  When I noticed that python 3.7 is not supported then the "is this thing dormant or active" comes into the picture. Then the list of Open Issues vs Closed Issues confirms this doubt:  in the past 18 months there have been dozens of issues opened but only five closed.

What is the prognosis for depending on this tool?

**Comment by javadba** [comment]
Oh I see the primary author switched focus to julia and `weave.jl` . Is anyone else likely to pick up the slack?

**Comment by slackline** [comment]
No idea, but I discovered [codebraid](https://github.com/gpoore/codebraid) the other day which may be of interest to you.

**Comment by metasim** [comment]
Is there some neutral organization under which we could fork this and keep it going[^1]? I primarily appreciate Pweave for the option of going from a `script.py` file to evaluated docs.

[^1]: I'm only a sometime Python dev (mostly Scala), so I can't take on the work myself.

**Comment by mpastell** [comment]
Sorry for the long delay in responding to this issue, I haven't had time to work on this or other projects lately. 

As I'm also not using Python actively at the moment I would be happy to transfer this project to an active organization. Let me know if you have a suggestion for a new home. Or I could create a new organization where other interested people could continue to maintain this project.

In addition to myself @abukaj and @piccolbo have commit access to the repository and might be interested in maintenance?

**Comment by piccolbo** [comment]
Unfortunately hit some major disagreement with other committer on first PR and couldn't find a way forward. But jupyter is the 800-pound gorilla. Vast majority of data people think it's ok to do work in a un-versionable binary format. I've been treated like a weirdo because I use a niche project instead of the new powerpoint. So I see two problems here 1)Governance: just promoting a couple of folks to committer role did not cut it 2)Competition: is it possible to compete with jupyter notebooks?

**Comment by metasim** [comment]
>  is it possible to compete with jupyter notebooks?

FWIW, I see this project as complement to Jupyter. We write examples in PyMarkdown and generate HTML for the web docs, and Notebooks as a part of our pre-packaged deployment option. It is a fantastic means of version ~controlling~ controllable documentation while providing multiple "deployment" options.

**Comment by javadba** [comment]
I second @metasim 's comment: I do not like to use jupyter for development but rather as a communications and documentation tool.  Driving notebooks directly from the code is the way to go.

**Comment by mpastell** [comment]
I think the main question is if anyone is interested enough to maintain the project. Probably you can't compete with Jupyter, but the project can still be useful. My main motivation for development was writing lecture slides and notes, but I'm not teaching actively in my current position and I mainly use Julia in my research.

**Comment by metasim** [comment]
@piccolbo Can you decode this a bit?

> Unfortunately hit some major disagreement with other committer on first PR and couldn't find a way forward.

@mpastell According to the `LICENSE.txt`, it appears you hold the copyright, but it's not under an [OSI approved license](https://opensource.org/licenses). Are you open to relicensing it under, say, Apache 2?

**Comment by amichuda** [comment]
Hi all, long time user, first time caller here. Just wanted to chime in. 

I'm a user of pweave and I don't think that we should consider this as necessarily something that can beat Jupyter, but it is a super useful project with its own use case. I think that it's biggest benefit is in making papers that can look publishable with minimal work and having to mess around with xml files and Jupyter metadata, or nbconvert as in Jupyter and ipypublish.

Pweave has made a splash in the blogosphere. and there was even a Spyder extension that utilized it for visualizing reports. 

I find the integration with vscode or atom a lot more elegant and straightforward.

My python knowledge is not up to snuff enough to maintain the project (I've only noodled with the code to see how easy it would be to integrate vscode style code cells in pypublish), but I would be interested in being someone's apprentice!

In any case, @mpastell, thank you so much for this awesome project and I hope you find Julia horrible one day and come back to Python!

**Comment by piccolbo** [comment]
@metasim you can check the PR from fgregg to see how things panned out. I have my opinion but since I was involved in it I think it's best kept to myself. PR was eventually withdrawn, I was for merging. I also tried to continue work on my fork but collaboration didn't work there either -- check the PRs again. I think finding a way forward is more important than assigning blame, but maybe there's something to be learned there.

**Comment by mpastell** [comment]
The license is BSD, I though that is OSI approved? But yes I'm open to changing the license if needed.

It's unlikely that I'll find Julia horrible ;)  I still do some work with Python too, but not enough to find motivation (or time) for maintaining Pweave.

**Comment by metasim** [comment]
> The license is BSD, I though that is OSI approved?

My apologies; Until now I didn't realize the BSD license (unlike Apache and MIT) doesn't actually have the word "BSD" in it, unless you use a [SPDX Identifier](https://spdx.org/using-spdx-license-identifier). Yes, it's OSI approved, so my bad.

**Comment by elikesprogramming** [comment]
any update on this?, is anyone taking the plunge?

**Comment by cortner** [comment]
bump?

**Comment by JamesOwers** [comment]
Hi all. Whilst I will not have time to maintain the package by myself, I'm very keen to help with active development - jupyter notebooks do not fill the hole for generating reproducible scientific papers that RMarkdown with KnirR did for me (4 years ago!).

Please let me know how I can help (even if it's just submitting PRs for issues I and others have raised).

**Comment by elikesprogramming** [comment]
> Hi all. Whilst I will not have time to maintain the package by myself, I'm very keen to help with active development - jupyter notebooks do not fill the hole for generating reproducible scientific papers that RMarkdown with KnirR did for me (4 years ago!).
> 
> Please let me know how I can help (even if it's just submitting PRs for issues I and others have raised).

same here. I also miss an RMarkdown in the python ecosystem to generate reproducible reports. PWeave is the closest I've found, but many features still missing. So I ended up crafting some of my own in internal packages (e.g. support for a floating table of contents using tocify, include interactive datatables in html reports, make it easier to include plotly interactive plots, among others). It is mostly quick-and-dirty code. But with active development, I'd be happy to help.

**Comment by slackline** [comment]
@JamesOwers and @elikesprogramming Not trying to detract from Pweave but as I mentioned above there is the [codebraid](https://github.com/gpoore/codebraid) package which may fit your needs.

I've not used it much myself, but it looks similar to RMarkdown (which I have used a lot in the past) and the project seems to still see activity.

**Comment by JamesOwers** [comment]
Thank you very much for that link - you may have just saved a lot of duplicate coding work! I will try it out.

**Comment by elikesprogramming** [comment]
yeah, I took `codebraid` for a spin some time ago and unfortunately it did not convince me, so I sticked to Pweave. Perhaps it was too early then and if they have active development, they may have had major progress since then and, admittedly, I haven't tried it again. So @JamesOwers if you try it and find it awesome, it'd be great if you give us here a heads up.

**Comment by elikesprogramming** [comment]
yeah, I took `codebraid` for a spin some time ago and unfortunately it did not convince me, so I sticked to Pweave. Perhaps it was too early then and if they have active development, they may have had major progress since then and, admittedly, I haven't tried it again. So @JamesOwers if you try it and find it awesome, it'd be great if you give us here a heads up.

**Comment by hdelrio** [comment]
I have never maintained any package, I am interested in helping in this project.

**Comment by contactzen** [comment]
@mpastell, 
I would like to take ownership of this project to keep ticking. Will you give me permission to use your code? Can we change the license to  MIT?

**Comment by contactzen** [comment]
I have took inspiration from peweave to but zen-knit. Follow at https://github.com/Zen-Reportz/zen_knit

**Comment by edalfon** [comment]
For those of you that used pweave because it was the closest alternative to R Markdown in python (and/or do not enjoy jupyter), now there's [Quarto https://quarto.org/](https://quarto.org/), which is very similar to R Markdown, but language agnostic and currently has integrations with "Python, R, and JavaScript via integration with Jupyter, Knitr, and Observable".

**Comment by dfolch** [comment]
I wanted to add a thank you for creating pweave. I really like it.

For those arriving here, the following is a summary of what I have found as of the end of 2023. To be clear, there is major overlap between the following three projects, so I'm only commenting on the differences from the perspective of writing academic papers.

- pweave
  - This is a great project but has been inactive for a few years.
- [Codebraid](https://codebraid.org/) ([github](https://github.com/gpoore/codebraid))
  - This is an active project run by seemingly one person. 
  - For my purposes, its biggest strength is its ability to interlace Markdown text and Python code.
  - I could not figure out how to do figure cross-referencing via pandoc-xnos.
- [Quarto](https://quarto.org/)
  - This is an active project run by the folks who brought you RStudio. 
  - For my purposes, its biggest strength is its seamless ability to do cross-references (e.g., table and figure numbering and integrating citations) without add ons. Also, great documentation.
  - Interlaced Python code is possible, but impractical. I will spare you the details, but you cannot have Python code and complicated Latex math in the same paragraph. Even if you never have Latex math together with Python code, interlaced Markdown and Python must be in a code chunk (not a Markdown chunk).
 
Since the latter two are under active development, I suspect the weaknesses will eventually be addressed. A very hacky solution in the mean time is to combine them.
- Run Codebraid first to process the interlaced Markdown and Python (basically, using Codebraid to process the Markdown chunks)
- Run Quarto on Codebraid's Markdown output to produce the final paper with rendered figures and cross-referencing (note: you need to manually switch some of Codebraid's syntax to Quarto syntax first)

---
__March 2024 Update__

Quarto 1.4 released in January 2024 now allows inline code in Markdown chunks. In my opinion, it's now a complete option for academic papers.
